projects/vdk-plugins/vdk-jupyter/vdk-jupyterlab-extension/src/serverRequests.ts

```diff
@@ -47,6 +63,66 @@ type jobRequestResult = {
   isSuccessful: boolean;
 };
 
+type TaskStatusResult = {
+  /**
+   * Type of the task
+   */
+  task_type: string;
+  /**
+   * Status of the task
+   */
+  status: string;
+  /**
+   * Result message of the task if no errors occurred
+   */
+  message: string | IJobConvertToNotebookMessage;
+  /**
+   * Error message
+   */
+  error: string;
+};
+
+const pollForTaskCompletion = async (
+  taskId: string,
+  maxAttempts = 43200,
+  interval = 10000,
+  errorThreshold = 5
+): Promise<TaskStatusResult> => {
+  let errorCounter = 0;
+  for (let i = 0; i < maxAttempts; i++) {
+    try {
+      const result = await requestAPI<TaskStatusResult>(
+        `taskStatus?taskId=${taskId}`,
+        { method: 'GET' }
+      );
+      if (
+        result.task_type === taskId &&
+        (result.status === 'completed' || result.error)
+      ) {
+        return result;
+      }
+    } catch (error) {
+      showError(error);
```

Comment:
Here we show the error but we keep trying to run the operation wouldn't it somehow confuse the user? 
Does the error say the VDK operation will continue running if not maybe it should be added as some message

Reply:
The argument is the same as above. If we don't find it reasonable enough to handle such cases using the error threshold, we can directly return the error on each occasion and stop the operation on the first error.